analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp8.cs

```diff
@@ -0,0 +1,209 @@
+﻿public class NullCoalescenceAssignment
+{
+    public void NullCoalescenceAssignment_NotNull(string s1)
+    {
+        s1 ??= "N/A";
+        s1.ToString(); // Compliant
+    }
+
+    public void NullCoalescenceAssignment_Null(string s1)
+    {
+        s1 ??= null;
+        s1.ToString(); // FIXME Non-compliant
+    }
+
+    public void InsideIf(string str)
+    {
+        if (str != null)
+        {
+            str ??= null;
+            str.ToString(); // Compliant, we know str is not null
+        }
+
+        if (str == null)
+        {
+            str ??= "foo";
+            str.ToString(); // Compliant, assigned foo
+        }
+    }
```

Comment:
These are S2259 test cases that probably shouldn't be here

Reply:
Removed it.